### PR TITLE
Philip M. 2108

### PR DESCRIPTION
**Pull request by PhiMed**

### CODE REVIEW

File: app/controllers/merchants_controller.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+class MerchantsController < ApplicationController
+  def index
+    @merchants = MerchantsFacade.all_merchants
+
+  end
+
+  def show
+    @merchant = MerchantsFacade.merchant_details(params[:id])
+    @items = MerchantsFacade.merchants_items(params[:id])
+  end
```
Comment (megstang) [inline app/controllers/merchants_controller.rb:10]: Nice job here. You could also use a MerchantItemsFacade, and then create MerchantItem objects to send back to the view.

File: app/facades/merchants_facade.rb [hunk 2]
```diff
@@ -0,0 +1,31 @@
+class MerchantsFacade
+  def self.find(name)
+    results = MerchantService.find_all(name)
+
+    if results != nil
+        results.map do |result|
+          Merchant.new(result)
+        end
+    end
```
Comment (megstang) [inline app/facades/merchants_facade.rb:9]: Another edge case you could test here is that if the data root key came back, that the array it was pointing to wasn't empty.

File: app/models/item.rb [hunk 3]
```diff
@@ -0,0 +1,10 @@
+class Item
+  attr_reader :name, :id, :description, :unit_price
+
+  def initialize(item)
+    @name = item[:attributes][:name]
+    @id = item[:id]
+    @description = item[:attributes][:description]
+    @unit_price = item[:attributes][:unit_price]
+  end
+end
```
Comment (megstang) [inline app/models/item.rb:10]: This goes for the merchant.rb poro as well, but be really careful naming an attribute "id", since all poros will automatically have a generated id on them. This could cause issues later on when you're trying to access the id that you manually stored here.

File: app/services/item_service.rb [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+  end
+
+  def self.parse_response(response)
+    JSON.parse(response, symbolize_names: true)
+  end
+
+  def self.conn
+    Faraday.new(url: "http://localhost:3000")
+  end
+end
```
Comment (megstang) [inline app/services/item_service.rb:25]: 👍

Comment (megstang) [reply app/services/item_service.rb:25]: This goes for the MErchantService too.. but don't go into the :data root key within the service. Let your service be responsible for calling and getting back ALL data -- If you had another facade that wanted data outside of the data root key, then you just limited that in the service. Let your facades manage the root keys that you need to filter through.

**Review by megstang — COMMENTED** [review]
@PhiMed Great job on the challenge!
